src/main/java/com/aws/greengrass/componentmanager/ComponentStore.java

```diff
@@ -496,7 +505,8 @@ public RecipeMetadata getRecipeMetadata(ComponentIdentifier componentIdentifier)
 
             throw new PackageLoadingException(String.format(
                     "Failed to get recipe metadata because the recipe metadata file should be a json but is corrupted."
-                            + " RecipeMetadataFilePath: '%s'.", metadataFile.getAbsolutePath()), e);
+                            + " RecipeMetadataFilePath: '%s'.", metadataFile.getAbsolutePath()), e)
+                    .withErrorContext(e, DeploymentErrorCode.RECIPE_METADATA_PARSE_ERROR);
```

Comment:
If it is just invalid JSON then just say something like: "Recipe metadata is not valid JSON: %s" ?